tests/cases/fourslash/codeFixMissingTypeAnnotationOnExports12.ts

```diff
@@ -0,0 +1,30 @@
+/// <reference path='fourslash.ts'/>
+
+// @isolatedDeclarations: true
+// @declaration: true
+//// function mixin<T extends new (...a: any) => any>(ctor: T): T {
+////     return ctor;
+//// }
+//// class Point2D { x = 0; y = 0; }
+//// export const Point3D = class extends mixin(Point2D) {  z = 0; };
+
+verify.codeFixAvailable([
+    { description: ts.Diagnostics.Declaration_emit_for_this_file_requires_type_resolution_An_explicit_type_annotation_may_unblock_declaration_emit.message },
+    { description: ts.Diagnostics.Declaration_emit_for_this_file_requires_type_resolution_An_explicit_type_annotation_may_unblock_declaration_emit.message }
+]);
+
+// TODO: There's no easy way to name the type, so rather promoting this to a classDeclaration is better.
+verify.codeFix({
+    description: ts.Diagnostics.Declaration_emit_for_this_file_requires_type_resolution_An_explicit_type_annotation_may_unblock_declaration_emit.message,
+    index: 0,
+    newFileContent:
+`function mixin<T extends new (...a: any) => any>(ctor: T): T {
+    return ctor;
+}
+class Point2D { x = 0; y = 0; }
```

Comment:
Why can't we use the same approach here as for class declarations? 

```ts
class Point2D { x = 0; y = 0; }
const Point3DBase: typeof Point2D = (mixin(Point2D));
export const Point3D = class extends Point3DBase {  z = 0; }`
});
```

Reply:
It's because the diagnostics occur on `Point3D`, not on the `extends` clause. 
I can try to make this pattern recognized by the fixer, but potentially having a diagnostic on Point3D and a nested information on extends. Either way, I'll try this out in following PRs.